### PR TITLE
Support deletion and deprecation of GCE's images.

### PR DESCRIPTION
**Pull request by fcuny**
Add a new class GCENodeImage to represent GCE's images, and support the
`delete` and `deprecated` functions.

Added a new extra field to the image object to specify the deprecation
status.

### CODE REVIEW

File: libcloud/compute/drivers/gce.py [hunk 1]
```diff
@@ -229,6 +229,38 @@ def __repr__(self):
             self.id, self.name, self.address)
 
 
+class GCENodeImage(NodeImage):
+    """A GCE Node Image class."""
+    def __init__(self, id, name, driver, extra=None):
+        self.extra = extra
```
Comment (Kami) [inline libcloud/compute/drivers/gce.py]: You already pass extra to the parent constructor which means you shouldn't need to set the attribute in the line above.

File: libcloud/compute/drivers/gce.py [hunk 2]
```diff
@@ -1854,6 +1885,65 @@ def ex_destroy_address(self, address):
...
+
+        if not hasattr(replacement, 'name'):
+            replacement = self.ex_get_image(replacement)
+
+        if state is None:
+            state = 'DEPRECATED'
+
+        possible_states = ['DELETED', 'DEPRECATED', 'OBSOLETE']
+
+        if state not in possible_states:
```
Comment (Kami) [inline libcloud/compute/drivers/gce.py:1931]: Should throw `ValueError` instead of returning `False`

File: libcloud/compute/drivers/gce.py [hunk 3]
```diff
@@ -1854,6 +1885,65 @@ def ex_destroy_address(self, address):
...
+            return False
+
+        image_data = {
+            'state':       state,
```
Comment (Kami) [inline libcloud/compute/drivers/gce.py]: IIRC, this will cause lint to fail (extra whitespace)

File: libcloud/compute/drivers/gce.py (hunk 2 above)
Comment (fcuny) [inline libcloud/compute/drivers/gce.py:1931]: Changed.

**Comment by fcuny** [comment]
I'll push an update for the build failure.

**Comment by Kami** [comment]
Merged into trunk.

Thanks.

**Pull request closed without merging**